pkg/controllers/metrics/pod/controller.go

```diff
@@ -207,20 +213,20 @@ func (c *Controller) recordPodStartupMetric(pod *corev1.Pod) {
 				podName:      pod.Name,
 				podNamespace: pod.Namespace,
 			})
-			PodStartupDurationSeconds.Observe(cond.LastTransitionTime.Sub(pod.CreationTimestamp.Time).Seconds(), nil)
+			PodStartupDurationSeconds.Observe(cond.LastTransitionTime.Sub(schedulableTime).Seconds(), nil)
 			c.pendingPods.Delete(key)
 		}
 	}
 }
-func (c *Controller) recordPodBoundMetric(pod *corev1.Pod) {
+func (c *Controller) recordPodBoundMetric(pod *corev1.Pod, schedulableTime time.Time) {
 	key := client.ObjectKeyFromObject(pod).String()
 	condScheduled, ok := lo.Find(pod.Status.Conditions, func(c corev1.PodCondition) bool {
 		return c.Type == corev1.PodScheduled
 	})
 	if pod.Status.Phase == phasePending {
```

Comment:
With the schedulable time tracking, does that mean that we get to drop all of the tracking around seeing the pod/seeing that the pod was in a pending state inside of this controller?